src/Cake.Common/IO/DirectoryAliases.cs

```diff
@@ -319,6 +320,49 @@ public static void EnsureDirectoryExists(this ICakeContext context, DirectoryPat
             }
         }
 
+        /// <summary>
+        /// Deletes the specified directory and its contents if it exists.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// EnsureDirectoryDoNotExists("./be/gone");
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The directory path.</param>
+        [CakeMethodAlias]
+        [CakeAliasCategory("DoNotExists")]
+        public static void EnsureDirectoryDoNotExists(this ICakeContext context, DirectoryPath path)
+        {
+            context.EnsureDirectoryDoNotExists(path, new DeleteDirectorySettings { Recursive = true, Force = true });
+        }
+
+        /// <summary>
+        /// Deletes the specified directory if it exists.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// EnsureDirectoryDoNotExists("./be/gone", new DeleteDirectorySettings {
+        ///     Recursive = true,
+        ///     Force = true
+        /// });
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The directory path.</param>
+        /// <param name="settings">The delete settings.</param>
+        [CakeMethodAlias]
+        [CakeAliasCategory("DoNotExists")]
+        public static void EnsureDirectoryDoNotExists(this ICakeContext context, DirectoryPath path, DeleteDirectorySettings settings)
```

Comment:
Same as above... We should name it `EnsureDirectoryDoesNotExist` to be grammatically correct

```suggestion
        public static void EnsureDirectoryDoesNotExist(this ICakeContext context, DirectoryPath path, DeleteDirectorySettings settings)
```

Reply:
`DeleteDirectorySettings` makes a lot of sense to me, but I'm thinking if we shouldn't (for consistency) add an overload that receives an `EnsureDirectoryDoesNotExistSettings` (new, to be added to this PR, inheriting from `DeleteDirectorySettings`) which is what devs used to Cake would expect.

We could keep this overload with `DeleteDirectorySettings` as well as it's a good.

```csharp
EnsureDirectoryDoesNotExist(DirectoryPath)
EnsureDirectoryDoesNotExist(DirectoryPath, DeleteDirectorySettings)
EnsureDirectoryDoesNotExist(DirectoryPath, EnsureDirectoryDoesNotExistSettings)
```

@cake-build/cake-team & @nikhilagrawaldotnet what do you think?

Reply:
@nikhilagrawaldotnet Let's go with adding the additional overload `EnsureDirectoryDoesNotExistSettings` if you don't mind and get this merged in. Everything else looks great as-is.

Reply:
@gep13 @devlead - FYI ^

Reply:
@nikhilagrawaldotnet Please ping us when you update the PR